deps/src/main/java/com/microsoft/azure/sdk/iot/deps/util/Tools.java

```diff
@@ -215,8 +215,8 @@ public static void appendJsonAttribute(StringBuilder strBuilder, String name, St
             // Codes_SRS_SDK_JAVA_TOOLS_12_023: [The function shall append the input StringBuilder string using the following format: “name”:”value” if isQuoted is false]
             strBuilder.append("\"");
 
-            if (Tools.isNullOrEmpty(name))
-                strBuilder.append("");
+            if (Tools.isNullOrEmpty(name)) {
+            }
```

Comment:
This if block doesn't do anything when the condition is met, so you can get rid of it